prober: check response http status code

markmc opened the pull request:
I got my peer and client URLs mixed up in my cluster config, and I was seeing:

```
  W | rafthttp: health check for peer XXXX could not connect: json: cannot unmarshal number into Go value of type probing.Health
```

turns out $endpoint/raft/probing was returning 404. By checking the response status code, we can improve that to:

```
  W | rafthttp: health check for peer XXXX could not connect: got unexpected HTTP status code 404 Not Found from http://$endpoint/raft/probing
```

markmc commented:
Fixes etcd-io/etcd/issues/10404

xiang90 commented on prober.go:
```diff
@@ -60,6 +61,10 @@ func (p *prober) AddHTTP(id string, probingInterval time.Duration, endpoints []s
 				}
 				resp, err := p.tr.RoundTrip(req)
+				if err == nil && resp.StatusCode != http.StatusOK {
+					err = fmt.Errorf("got %s from %s", resp.Status, endpoints[pinned])
```
got unexpected HTTP status code xxx from xxx.

markmc replied:
Good point, updated now. Thanks for the review.

markmc committed:
prober: check response http status code

I got my peer and client URLs mixed up in my cluster config, and I was
seeing:

  W | rafthttp: health check for peer XXXX could not connect: json: cannot unmarshal number into Go value of type probing.Health

turns out $endpoint/raft/probing was returning 404. By checking the
response status code, we can improve that to:

  W | rafthttp: health check for peer XXXX could not connect: got unexpected HTTP status code 404 Not Found from http://$endpoint/raft/probing
